src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -382,16 +381,6 @@ private void OnBatchScopeDisposed()
                     return solution;
                 });
 
-                foreach (var (documentId, textContainer) in documentsToOpen)
-                {
-                    _workspace.ApplyChangeToWorkspace(w => w.OnDocumentOpened(documentId, textContainer));
```

Comment:
As discussed earlier: we do still want this for the classic contained language case, so best to just add a flag I think.

Reply:
sure. I will add a dictionary that tracks the open state.